src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -173,6 +175,18 @@ class Decimal extends React.Component {
       return this.state.value;
     }
 
+    const numberFormat = {
+      number: {
+        format: {
+          delimiter: this.props.delimiter,
+          separator: this.props.separator,
+        },
+      },
+    };
+
+    I18n.locale = this.props.locale;
+    I18n.translations[I18n.locale] = numberFormat;
```

Comment:
Why are we overwriting the translations for the locale? That's not something this component should be doing.

if the component receives `locale` then we should use I18n.t to get the format. Or we remove local and we just use the provided props.

Reply:
@chrisbarber86  - removing usage of I18n.locale requires also changes for toStandardDecimal - just using delimiter and separator from props in formatDecimal functions cause change of hiddenValue. Do you really think that'll be better in that case?

Reply:
my point was you are changing the value of I18n.locale AND I18n.translations[I18n.locale] - both of which are defined and owned by the application - this component has no right to override them. A requirement of using I18n here would be that the consuming application knows to setup:

```javascript
    {
      number: {
        format: {
          delimiter: ",",
          separator: ".",
        },
      }
    }
```

Reply:
In that case there is any better option to decide how number will be formatted? Right now it works according to browser settings and can't be changed, am I right?